go/fn/testhelpers/golden.go

```diff
@@ -75,33 +81,39 @@ func RunGoldenTests(t *testing.T, basedir string, krmFunction fn.ResourceListPro
 
 			var functionConfig *fn.KubeObject
 			if len(config) == 0 {
-				functionConfig = nil
+				functionConfig = fn.NewEmptyKubeObject()
 			} else if len(config) == 1 {
 				functionConfig = config[0]
 			} else {
 				t.Fatalf("found multiple config objects in %s", filepath.Join(dir, "_fnconfig.yaml"))
 			}
-
 			rl := &fn.ResourceList{Items: items, FunctionConfig: functionConfig}
-			success, err := krmFunction.Process(rl)
+			_, err = krmFunction.Process(rl)
```

Comment:
`success` tells the kpt CLI to treat function as pass or not. A golden test should accept "FAIL" Process, and in such case, the `resourcelist.results` would contain the detailed error messages.
```
[PASS]: ...
``` 
or 
```
[FAIL]: ...
```